geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/io/hadoop/HadoopGeoTiffRDDWrapper.scala

```diff
@@ -48,25 +41,26 @@ object HadoopGeoTiffRDDWrapper {
     keyType: String,
     path: String,
     sc: SparkContext
-  ): (JavaRDD[Array[Byte]], String) =
+  ): RasterRDD[_] =
     keyType match {
       case "ProjectedExtent" =>
-        PythonTranslator.toPython(HadoopGeoTiffRDD.spatialMultiband(path)(sc))
+        new ProjectedRasterRDD(HadoopGeoTiffRDD.spatialMultiband(path)(sc))
       case "TemporalProjectedExtent" =>
-        PythonTranslator.toPython(HadoopGeoTiffRDD.temporalMultiband(path)(sc))
+        new TemporalRasterRDD(HadoopGeoTiffRDD.temporalMultiband(path)(sc))
     }
 
   def getRDD(
     keyType: String,
     path: String,
     options: java.util.Map[String, Any],
-    sc: SparkContext): (JavaRDD[Array[Byte]], String) = {
+    sc: SparkContext
+  ): RasterRDD[_] = {
     val hadoopOptions = HadoopGeoTiffRDDOptions.setValues(options)
     keyType match {
       case "ProjectedExtent" =>
-        PythonTranslator.toPython(HadoopGeoTiffRDD.spatialMultiband(path, hadoopOptions)(sc))
+        new ProjectedRasterRDD(HadoopGeoTiffRDD.spatialMultiband(path, hadoopOptions)(sc))
       case "TemporalProjectedExtent" =>
-        PythonTranslator.toPython(HadoopGeoTiffRDD.temporalMultiband(path, hadoopOptions)(sc))
+        new TemporalRasterRDD(HadoopGeoTiffRDD.temporalMultiband(path, hadoopOptions)(sc))
```

Comment:
Lets name this `GeoTiffRDD` and have it decide to use `S3GeoTiffRDD` or `HadoopGeoTiffRDD` based on the URI parameter. `getRDD` can be `get` so client code looks like `GeoTiffRDD.get("spatial", "s3://..", {})`